### PR TITLE
Fix issue when using 'g' on regex

### PR DESCRIPTION
**Pull request by pajarraco**
Sorry for the formatting changes.

I found some issues when I use the 'g' option on regex. the last index need to be reset

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
   "peerDependencies": {
     "eslint": "*"
   },
-  "author": "Chris Banks <github@christopherbradleybanks.com>",
+  "author": "Pajarraco <https://github.com/pajarraco>",
```
Comment (christopherbradleybanks) [inline package.json]: Lets not change the author field of the package json
instead add "contributors": [{
  "name" : "Barney Rubble",
  "email" : "b@rubble.com",
  "url" : "http://barnyrubble.tumblr.com/"
}] to the package JSON
 
 or 
 
 [
"Barney Rubble <b@rubble.com> (http://barnyrubble.tumblr.com/)",
 ]

File: package.json [hunk 2]
```diff
@@ -21,6 +21,13 @@
...
     "type": "git",
-    "url": "git://github.com/christopherbradleybanks/eslint-plugin-folders.git"
+    "url": "git+ssh://git@github.com/pajarraco/eslint-plugin-folders.git"
+  },
+  "bugs": {
+    "url": "https://github.com/pajarraco/eslint-plugin-folders/issues"
+  },
+  "homepage": "https://github.com/pajarraco/eslint-plugin-folders#readme",
+  "directories": {
+    "lib": "lib"
```
Comment (christopherbradleybanks) [inline package.json]: this should not be included in the change

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "eslint-plugin-folders",
-  "version": "1.0.3",
+  "name": "eslint-plugin-folders2",
+  "version": "0.0.1",
```
Comment (christopherbradleybanks) [inline package.json]: lets not rename the plugin or reset the version

**Review by christopherbradleybanks — CHANGES_REQUESTED** [review]
Could you include a relevant example of what you are trying to test with this change thanks

File: package.json (hunk 1 above)
Comment (pajarraco) [reply package.json]: Hi, I was running this 
'folders/match-regex': [2, /^(?!\s)(?![\s\S]*\s$)[a-zA-Z0-9\s()-_&\%\.\/]+$/g, '/src/'], 

and noting was matching the expression, then I found this 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test#using_test_on_a_regex_with_the_global_flag

here they explain what happens with the index on "regex.test('foo', 'g')".

**Comment by pajarraco** [comment]
Sorry again for all the changes, now I create a more clean PR

File: package.json (hunk 3 above)
Comment (pajarraco) [reply package.json]: I changed this

File: package.json (hunk 2 above)
Comment (pajarraco) [reply package.json]: I changed back

**Review by christopherbradleybanks — APPROVED** [review]

**Review by christopherbradleybanks — APPROVED** [review]
